File: XMLCheck/Handler/CheckRulesXML08/Xml8RuleBase.cs
using XMLCheck.Models;
using XMLCheck.Services;

namespace XMLCheck.Handler.CheckRulesXML08
{
    public abstract class Xml8RuleBase:IXml8Rule
    {
        public abstract string Key { get; }

        public abstract ErrorDetails? Check(Xml8Model model,HoSoContext context);

        protected ErrorDetails Error(string message)
        {
            return new ErrorDetails
            {
                key = Key,
                message = message
            };
        }
    }
}

File: XMLValidator_v2/src/parsers/rules/xml8/xml8RuleBase.ts
import { Xml8Model } from '../../../interfaces/xml-models.interface';
import { HoSoContext } from '../../../services/hoso-context';
import { ErrorDetails } from '../../../dto/xml-check.dto';
import { BaseRule } from '../base.rule';

export interface IXml8Rule {
  readonly key: string;
  readonly isEnabled: boolean;
  check(model: Xml8Model, context: HoSoContext): ErrorDetails | null;
}

export abstract class Xml8RuleBase extends BaseRule<Xml8Model> implements IXml8Rule {}
